pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotClientRequest.java

```diff
@@ -559,6 +563,16 @@ public static Response getPinotQueryResponse(BrokerResponse brokerResponse)
     if (!exceptions.isEmpty()) {
       // set the header value as first exception error code value.
       queryErrorCodeHeaderValue = exceptions.get(0).getErrorCode();
+
+      // do log with the exception flagged with a particular marker for filtering
+      MDC.put("queryErrorCode", Integer.toString(queryErrorCodeHeaderValue));
+      StringBuilder sb = new StringBuilder();
+      sb.append("Query processing exceptions:");
+      for (QueryProcessingException exception : exceptions) {
+        sb.append(" ").append(exception.toString());
+      }
```

Comment:
Do we also need to add it into the MDC? Markers are not enough to route the error?

Anyway, can you modify the PR title/description?